inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -67,5 +91,98 @@ public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel,
                 aAnnotationPage.actionRefreshDocument(_target)));
         
         add(form);
-    }
+
+		chartContainer = new Label("chart-container" );
+		add(chartContainer);
+
+		add(new AbstractAjaxTimerBehavior(Duration.milliseconds(5000)) {
+			private static final long serialVersionUID = -3782208159226605584L;
+
+			@Override
+			protected void onTimer(AjaxRequestTarget target) {
+
+				List<LoggedEvent> loggedEvents = recommendationService.listLoggedEvents(aModel.getObject().getProject(),
+						aModel.getObject().getUser());
+
+				// iterate over the logged events to extract the scores only.
+				List<Double> listDetailJson = new ArrayList<Double>();
```

Comment:
If we know the size of the list to start with, I'd recommend to use an array of double (`double[]`) instead of a collection class.

Reply:
Sometimes the score in the database is "NAN". Should we plot it as 0 or skip it? In case of skipping, we dont know the size of the array of valid scores to start with.

Reply:
I have removed the check for 0 - how does the chart behave if there is a NAN in between?

Reply:
I'd say probably render it as 0 if the chart fails treating the NAN natively.

Reply:
It ignores the NaN value and does not plot the point on the graph

Reply:
Sounds ok to me for the time being.